### PR TITLE
add share page and buttons

### PR DESCRIPTION
**Pull request by e91096**
[Trello Card](https://trello.com/c/Q7Nud5u2/71-a-constituent-should-see-an-invite-page)

screen shot 2018-10-12 at 2 59 45 am

### CODE REVIEW

**Comment by chrisruenes1** [comment]
@e91096 this is so cool!!!! It makes PL seem really *real*

I left a couple small comments in the code, mostly about the shape of the `currentCampaign` data the store.

A couple other small bugs I noticed:

I at first thought this was cause I wasn't logged in, but after I created an account this was still happening: 
screen shot 2018-10-03 at 10 15 09 pm

Also, this is _really_ small, but do we have any control over how these links render before they get decorated with the social media css? On my very slow computer with my very slow wifi, I'm getting a bit of time where they are just appearing as links:

screen shot 2018-10-03 at 10 32 05 pm

I saw your comment about sharing pre-written messages in facebook—are we gonna leave that for now? Totally fine by me if we are, but I wanted to make sure it wasn't a Not-clearing-errors-on-navigation-cause-I-forgot situation!

Ok otherwise this is awesome!! Thanks for doing it!!!

**Comment by e91096** [comment]
> I at first thought this was cause I wasn't logged in, but after I created an account this was still happening:

ooooh well that definitely doesn't seem right! great catch i'll look into it!

> Also, this is really small, but do we have any control over how these links render before they get decorated with the social media css? On my very slow computer with my very slow wifi, I'm getting a bit of time where they are just appearing as links:

ah interesting haha ¯\_(ツ)_/¯ i can probably style the links before the social media css takes over? what do you think they should look like?

> I saw your comment about sharing pre-written messages in facebook—are we gonna leave that for now? Totally fine by me if we are, but I wanted to make sure it wasn't a Not-clearing-errors-on-navigation-cause-I-forgot situation!

i'm not sure if you looked into it at all (i haven't dug any further) but my suspicion is that they don't surface an easy way to do it without a facebook app id...

**Comment by chrisruenes1** [comment]
Hey @e91096 do you think we could just hide the links until they are styled?

File: frontend/components/Navbar.jsx [hunk 1]
```diff
@@ -53,35 +56,41 @@ class Navbar extends Component {
...
       if (this.props.currentUser.admin) {
         NAV_ITEMS_ADMIN.forEach((item) => {
-          navItems.push(
-            <NavItem
-              {...item}
-              handleClick={() => { this.handleNavClick(item); }}
-              selected={isItemSelected(item.path)}
-              key={item.name}
-            />);
+          navItems.push(<NavItem
```
[the remark this answers is not in the document]
Comment (e91096) [reply frontend/components/Navbar.jsx]: yup! i was messing around with my linter at work and i had it turned off for a minute but it's back now and as opinionated as ever

File: frontend/components/Navbar.jsx [hunk 2]
```diff
@@ -102,10 +111,12 @@ class Navbar extends Component {
 }
 
 const mapDispatchToProps = dispatch => ({
+  fetchCurrentCampaign: () => dispatch(fetchCurrentCampaign()),
   logoutUser: (handleSuccess, handleFail) => dispatch(logoutUser(handleSuccess, handleFail)),
 });
 
 const mapStateToProps = state => ({
+  currentCampaign: state.users.currentCampaign,
```
[the remark this answers is not in the document]
Comment (e91096) [reply frontend/components/Navbar.jsx]: ooooooh. 😳

File: frontend/reducers/campaignsReducers/byId.js [hunk 3]
```diff
@@ -12,6 +12,11 @@ export default function byId(state = {}, action) {
       };
     case 'RECEIVE_CAMPAIGNS':
       return nest(payload.campaigns, 'id');
+    case 'RECEIVE_CURRENT_CAMPAIGN':
+      return {
+        ...state,
+        currentCampaign: payload.campaign,
```
[the remark this answers is not in the document]
Comment (e91096) [reply frontend/reducers/campaignsReducers/byId.js]: ooooh yeah i agree that's a great idea

File: frontend/components/SharePage.jsx [hunk 4]
```diff
@@ -1,11 +1,18 @@
 import React from 'react';
 import { connect } from 'react-redux';
 import * as _ from 'underscore';
+import {
```
Comment (chrisruenes1) [inline frontend/components/SharePage.jsx]: Niiiiice!

File: frontend/components/LoginForm.jsx [hunk 5]
```diff
@@ -69,13 +71,10 @@ class LoginForm extends Component {
...
 
   render() {
-    if (this.props.currentUser) {
-      return (
-        <div>
-          You already did this, you silly goose. Go <a href="/success" onClick={push}>here</a>
-        </div>
-      );
+    if (!_.isEmpty(this.props.currentUser)) {
+      setTimeout(() => { push('/success'); }, 0);
```
Comment (chrisruenes1) [inline frontend/components/LoginForm.jsx]: Say @e91096 I don't get it! Why would I want to make something a timeout if it's gonna time out at 0 seconds? (I am sure there is a reason for doing this I don't know but wanted you to know how I feel when I see it—ok thanks!)

File: frontend/components/Navbar.jsx [hunk 6]
```diff
@@ -24,8 +27,8 @@ class Navbar extends Component {
 
   componentWillMount() {
     document.addEventListener('mousedown', this.handleClick, false);
-    // TODO fetch current user
     this.props.fetchCurrentCampaign();
+    this.props.fetchCurrentUser();
```
Comment (chrisruenes1) [inline frontend/components/Navbar.jsx]: weee!

File: frontend/components/Navbar.jsx [hunk 7]
```diff
@@ -47,12 +50,12 @@ class Navbar extends Component {
   }
 
   render() {
-    const isItemSelected = (pathName) => {
+    const isItemSelected = (pathName) => { // TODO isItemSelected not used. use it probably
```
Comment (chrisruenes1) [inline frontend/components/Navbar.jsx]: hahaha wait should we just take this out?

File: frontend/components/Navbar.jsx [hunk 8]
```diff
@@ -120,4 +124,18 @@ const mapStateToProps = state => ({
   currentUser: state.users.currentUser,
 });
 
+Navbar.propTypes = {
```
Comment (chrisruenes1) [inline frontend/components/Navbar.jsx]: proptypes!

File: frontend/components/PasswordResetForm.jsx [hunk 9]
```diff
@@ -1,9 +1,12 @@
 import React, { Component } from 'react';
```
Comment (chrisruenes1) [inline frontend/components/PasswordResetForm.jsx]: Ok in one of my branches, I renamed this file to `RequestPasswordResetForm`...would you mind doing that here too? I think that if we both make changes to this file, but mine is actually a different file, that could lead to some very cruel rebase mistakes

File: frontend/components/SuccessStub.jsx [hunk 10]
```diff
@@ -1,27 +1,35 @@
 import React from 'react';
```
Comment (chrisruenes1) [inline frontend/components/SuccessStub.jsx]: Is this still being used?

File: frontend/reducers/campaignsReducers/currentCampaign.js [hunk 11]
```diff
@@ -0,0 +1,9 @@
+export default function currentCampaign(state = null, action) {
```
Comment (chrisruenes1) [inline frontend/reducers/campaignsReducers/currentCampaign.js]: Hey @e91096 I have one more level of suggestion about how to make this a little more idiomatic redux-y, but I don't think that should be a reason not to merge this in. I can show you what I'm thinking next time we hack together!

**Pull request merged**